src/textual_fspicker/base_dialog.py

```diff
@@ -162,7 +175,16 @@ def _set_error(self, message: str = "") -> None:
     @on(DriveNavigation.DriveSelected)
     def _change_drive(self, event: DriveNavigation.DriveSelected) -> None:
         """Reload DirectoryNavigation in response to drive change."""
-        self.query_one(DirectoryNavigation).location = event.drive_root
+        """Reload DirectoryNavigation in response to drive change."""
+        dir_nav = self.query_one(DirectoryNavigation)
+        dir_nav.location = event.drive_root
+
+    @on(DirectoryNavigation.Changed)
+    def _on_directory_changed(self, event: DirectoryNavigation.Changed) -> None:
```

Comment:
There's already an event handler for `DirectoryNaviagion.Changed` in the change named `_clear_error`; this new event handler seems to be a) named so that it doesn't need to use `@on` anyway and b) to do the same thing as the original error handler, plus more.

This is quite confusing and probably results on more work than is necessary.

You might want to review what it is you're trying to do here.